connector/introspection/introspection.go

```diff
@@ -0,0 +1,361 @@
+		groupsKey:                 c.ClaimMapping.GroupsKey,
+		newGroupFromClaims:        c.ClaimMutations.NewGroupFromClaims,
+		groupsFilter:              groupsFilter,
+	}, nil
+}
+
+var (
+	_ connector.TokenIdentityConnector = (*introspectionConnector)(nil)
+)
+
+type introspectionConnector struct {
+	introspectionURL          string
+	clientID                  string
+	clientSecret              string
+	scopes                    []string
+	logger                    *slog.Logger
+	httpClient                *http.Client
+	insecureSkipEmailVerified bool
+	insecureEnableGroups      bool
+	allowedGroups             []string
+	userIDKey                 string
+	userNameKey               string
+	overrideClaimMapping      bool
+	preferredUsernameKey      string
+	emailKey                  string
+	groupsKey                 string
+	newGroupFromClaims        []NewGroupFromClaims
+	groupsFilter              *regexp.Regexp
+}
+
+func (c *introspectionConnector) Close() error {
+	return nil
+}
+
+func (c *introspectionConnector) Type() string {
+	return "introspection"
+}
+
+func (c *introspectionConnector) TokenIdentity(ctx context.Context, subjectTokenType, subjectToken string) (connector.Identity, error) {
+	// Create form data for the introspection request
+	form := url.Values{}
+	form.Add("token", subjectToken)
+	form.Add("token_type_hint", subjectTokenType)
+
+	// Create the request
+	req, err := http.NewRequestWithContext(ctx, "POST", c.introspectionURL,
+		strings.NewReader(form.Encode()))
+	if err != nil {
+		return connector.Identity{}, fmt.Errorf("introspection: failed to create request: %v", err)
+	}
+
+	// Set headers
+	req.Header.Add("Content-Type", "application/x-www-form-urlencoded")
+
+	// Add client authentication if provided
+	if c.clientID != "" && c.clientSecret != "" {
+		req.SetBasicAuth(c.clientID, c.clientSecret)
+	}
+
+	// Send the request
+	resp, err := c.httpClient.Do(req)
+	if err != nil {
+		return connector.Identity{}, fmt.Errorf("introspection: request failed: %v", err)
+	}
+	defer resp.Body.Close()
+
+	// Check the response status
+	if resp.StatusCode != http.StatusOK {
+		return connector.Identity{}, fmt.Errorf("introspection: endpoint returned status %d", resp.StatusCode)
+	}
+
+	// Parse the response
+	var introspectionResp struct {
+		Active bool                   `json:"active"`
+		Extra  map[string]interface{} `json:"-"`
+	}
+
+	// Use a decoder that captures unknown fields
+	decoder := json.NewDecoder(resp.Body)
+	if err := decoder.Decode(&introspectionResp); err != nil {
+		return connector.Identity{}, fmt.Errorf("introspection: failed to decode response: %v", err)
+	}
+
+	// Check if token is active
+	if !introspectionResp.Active {
+		return connector.Identity{}, errors.New("introspection: token is not active")
+	}
+
+	// Now extract the claims from the response
+	var claims map[string]interface{}
+	if err := json.NewDecoder(strings.NewReader(resp.Header.Get("X-Debug"))).Decode(&claims); err != nil {
```

Comment:
we don't get X-Debug in response header, this may throw an error if not a valid json.
If `resp.Header.Get("X-Debug")` is empty, use `{}`